feat: added tests for views

Asespinel opened the pull request:
This PR adds basic testing for the happy path to submit a course for validation until it gets approved by a validator. It also adds a test to check the case where a validation body is exempt to skip the normal validation process.

**How to test**

1. Inside your platform-global-teacher-campus folder install a virtual environment
2. Run the command virtualenv -p python3.8 venv
3. Activate the environment by using source venv/bin/activate
4. Install the requirements by running pip install -r requirements/test.txt
5. Create an env variable for tox: export TOX_ENV=py38-django32
6.  Set the newly created varaible export TOXENV=${TOX_ENV//./}
7. With your activated environmment run tox
8. You should see the 3 tests passing.

MaferMazu commented:
Thanks for this PR, @Asespinel 🙌
I have only two questions.

- Can we change the Bearer authentication to JWT auth? Is it possible, or does that break all? I ask this because we are using JWT in our views, and it makes sense to use JWT in the tests.

- Can we use data more realistically for the CourseOverviewTestModel? In the CourseOverview model, the id is not a numerical id; it is a CourseKey, https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/models.py#L74, maybe not use a CourseKey but at least we can use a strings as primary_key, e.g. "course-v1:edX+DemoX+Demo_Course". Adding to that question, I think it's unnecessary to declare self.course_id and self.course_id_exempt, you can access that info using the self.course.id and self.course_exempt.id; or if you want to define those variables, then use it to create the self.course and self.course_exempt.

The rest looks good to me.

Asespinel commented:
Thanks for the comments @MaferMazu, regarding your questions:

1. Yes if I explicitly use 'jwt' instead of 'bearer' the request won't load the jwt token in the payload and the tests couldn't access the endpoints properly.
2. I followed your suggestions and I changed the id into a tring type, and changed the course creations to use the defined variables. 

Let me know if you have any additional comments.

MaferMazu reviewed (approved):
It looks good.

It's a shame we couldn't use JWT as a prefix for the auth, but we already tested that we are using a JWT token to authenticate. For me, it's okay.
